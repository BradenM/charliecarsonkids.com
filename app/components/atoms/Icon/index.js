/**
 *
 * Icon
 *
 * Atom
 */

import PropTypes from 'prop-types';
import React from 'react';
import { ReactSVG } from 'react-svg';
import { IconURL } from 'utils/getStatic';

const Icon = ({ name, iconSize, width, ...props }) => (
  <ReactSVG
    beforeInjection={svg =>
      svg.setAttribute(
        'style',
        `width: ${width || iconSize}px;height: ${width || iconSize}px`,
      )
    }
    src={IconURL(name, iconSize)}
    {...props}
  />
);

Icon.defaultProps = {
  iconSize: '100',
  width: undefined,
};

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  iconSize: PropTypes.oneOf(['50', '100', '500', '32', '24']),
  width: PropTypes.string,
};

export default Icon;
