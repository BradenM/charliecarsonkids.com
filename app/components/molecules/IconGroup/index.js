/**
 *
 * IconGroup
 *
 * Molecule
 */

import PropTypes from 'prop-types';
import React from 'react';
import ListGroup from 'react-bootstrap/ListGroup';
import styled from 'styled-components';

const BorderlessItem = styled(ListGroup.Item)`
  border: 0;
  padding: 0 0.4em 0 0.4em;
  background-color: transparent;
`;

const IconGroup = ({ children, containerClass, ...props }) => (
  <ListGroup horizontal className={`border-0 ${containerClass}`}>
    {children.map((child, i) => (
      // eslint-disable-next-line react/no-array-index-key
      <BorderlessItem {...props} key={`iconchild_${i}`}>
        {child}
      </BorderlessItem>
    ))}
  </ListGroup>
);

IconGroup.defaultProps = {
  containerClass: '',
};

IconGroup.propTypes = {
  children: PropTypes.arrayOf(PropTypes.element).isRequired,
  containerClass: PropTypes.string,
};

export default IconGroup;
