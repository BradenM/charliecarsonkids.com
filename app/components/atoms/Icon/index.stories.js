/**
 *
 * Stories for Icon
 *
 * Atom
 */

import { text } from '@storybook/addon-knobs';
import React from 'react';
import Icon from '.';

export default {
  title: 'Design|Atoms/Icon',
  component: Icon,
};

export const facebookIcon = () => (
  <Icon
    name={text('Icon Name', 'facebook')}
    width={text('Width', '100')}
    iconSize="100"
  />
);
