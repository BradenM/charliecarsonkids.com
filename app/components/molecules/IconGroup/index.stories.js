/**
 *
 * Stories for IconGroup
 *
 * Molecule
 */

import Icon from 'atoms/Icon';
import IconButton from 'atoms/IconButton';
import React from 'react';
import IconGroup from '.';

export default {
  title: 'Design|Molecules/IconGroup',
  component: IconGroup,
};

export const socialIconGroup = () => (
  <IconGroup>
    <Icon name="facebook" width="50" />
    <Icon name="instagram" width="50" />
    <Icon name="mail" width="50" />
  </IconGroup>
);

export const socialIconButtonGroup = () => (
  <IconGroup>
    <IconButton name="facebook" url="https://facebook.com" />
    <IconButton name="instagram" url="https://instagram.com" />
    <IconButton name="mail" url="https://mail.com" />
  </IconGroup>
);
