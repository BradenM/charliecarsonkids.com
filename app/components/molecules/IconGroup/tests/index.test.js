/**
 *
 * Tests for IconGroup
 *
 * Molecules
 */

import Icon from 'atoms/Icon';
import React from 'react';
import { render } from 'react-testing-library';
// import 'jest-dom/extend-expect'; // add some helpful assertions
import IconGroup from '../index';

describe('<IconGroup />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <IconGroup>
        <Icon name="icon" />
        <Icon name="icon" />
      </IconGroup>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IconGroup>
        <Icon name="icon" />
        <Icon name="icon" />
      </IconGroup>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
