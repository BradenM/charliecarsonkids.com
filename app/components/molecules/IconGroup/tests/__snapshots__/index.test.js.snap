// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconGroup /> Should render and match the snapshot 1`] = `
<div
  class="border-0  list-group list-group-horizontal"
>
  <div
    class="IconGroup__BorderlessItem-sc-1br5l8c-0 gWwTrs list-group-item"
  >
    <div>
      <div>
        <div
          data-src="https://static.charliecarsonkids.com/assets/icons/icon-100.svg"
          src=""
        />
      </div>
    </div>
  </div>
  <div
    class="IconGroup__BorderlessItem-sc-1br5l8c-0 gWwTrs list-group-item"
  >
    <div>
      <div>
        <div
          data-src="https://static.charliecarsonkids.com/assets/icons/icon-100.svg"
          src=""
        />
      </div>
    </div>
  </div>
</div>
`;
